### PR TITLE
update getSetting

### PR DESCRIPTION
**Pull request by mssfang**
Updated getSetting sync and async APIs and all samples and tests

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 1]
```diff
@@ -360,48 +361,28 @@
...
-     * key is an invalid value (which will also throw HttpResponseException described below).
-     * @throws NullPointerException If {@code setting} is {@code null}.
-     * @throws IllegalArgumentException If {@link ConfigurationSetting#getKey() key} is {@code null}.
-     * @throws ResourceNotFoundException If a ConfigurationSetting with the same key and label does not exist.
-     * @throws HttpResponseException If the {@code} key is an empty string.
-     */
-    @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<ConfigurationSetting> getSetting(ConfigurationSetting setting) {
-        return withContext(context -> getSetting(setting, context))
+    public Mono<ConfigurationSetting> getSetting(String key, String label, OffsetDateTime acceptDateTime) {
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: Are we going to rename the OffsetDateTime to what .NET has? I believe it is called asOf

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java [hunk 2]
```diff
@@ -327,8 +333,8 @@ public ConfigurationSetting getSetting(String key) {
      * @throws HttpResponseException If the {@code} key is an empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public ConfigurationSetting getSetting(ConfigurationSetting setting) {
-        return getSetting(setting, Context.NONE).getValue();
+    public ConfigurationSetting getSetting(ConfigurationSetting setting, boolean onlyIfChanged) {
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java]: Remove this in favor of the maximal overload.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java [hunk 3]
```diff
@@ -295,30 +297,34 @@ public ConfigurationSetting updateSetting(ConfigurationSetting setting) {
...
      * @return The {@link ConfigurationSetting} stored in the service, or {@code null}, if the configuration value does
      * not exist or the key is an invalid value (which will also throw ServiceRequestException described below).
      * @throws IllegalArgumentException If {@code key} is {@code null}.
      * @throws ResourceNotFoundException If a ConfigurationSetting with {@code key} does not exist.
      * @throws HttpResponseException If {@code key} is an empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public ConfigurationSetting getSetting(String key) {
-        return getSetting(new ConfigurationSetting().setKey(key), Context.NONE).getValue();
+    public ConfigurationSetting getSetting(String key, String label, OffsetDateTime acceptDateTime) {
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java]: Follow-up with the group on whether we want the DateTime overload to be the simple overload. The date time use case is a bit more advanced.

Additionally, we need to have this parameter in the maximal overload. It should allow the same functionality as this one.

Comment (alzimmermsft) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java]: @JonathanGiles  Mind giving some guidance here, having the DateTime in the maximal overload would diverge from what the other languages have.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 4]
```diff
@@ -360,48 +361,28 @@
...
      *
-     * <p>Retrieve the setting with the key "prodDBConnection".</p>
+     * <p>Retrieve the setting with the key "prodDBConnection" and a time that one minute before now at UTC-Zone</p>
      *
-     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.getSetting#string}
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.getSetting#string-string-OffsetDateTime}
      *
      * @param key The key of the setting to retrieve.
+     * @param label Optional, the label of the setting to retrieve.
+     * @param acceptDateTime Optional, to access a past state of the configuration setting.
```
Comment (JonathanGiles) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: All of these parameters that are optional should not have their parameter description start by saying 'Optional'. It doesn't read well and nor does it speak to what the user should provide if they don't want to provide anything - null, empty string, etc? I would have this formatted to say "The label of the setting to retrieve, or optionally, null if a setting with label is desired" - or whatever the actual description text should say.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java (hunk 3 above)
Comment (JonathanGiles) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java]: I don't mind have multiple 'maximal' overloads. To me, it seems like having one that takes (key, label) is a minimal and (key, label, asOf) is a local maximal.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 5]
```diff
@@ -77,19 +78,19 @@
      * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.addSetting#string-string-string}
      *
      * @param key The key of the configuration setting to add.
-     * @param value The value associated with this configuration setting key.
-     * @param label The label of the configuration setting to create or update, or optionally, null if a setting with
+     * @param label The label of the configuration setting to add, or optionally, null if a setting with
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: Missing no in the case of null label

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 6]
```diff
@@ -240,53 +241,36 @@
...
-     * <p>Retrieve the setting with the key "prodDBConnection".</p>
+     * <p>Retrieve the setting with the key "prodDBConnection" and a time that one minute before now at UTC-Zone</p>
      *
-     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.getSetting#string}
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.getSetting#string-string-OffsetDateTime}
      *
      * @param key The key of the setting to retrieve.
+     * @param label The label of the configuration setting to retrieve, or optionally, null if a setting with
+     * label is desired.
+     * @param asOfDateTime To access a past state of the configuration setting, or optionally, null if a setting with
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]:
```suggestion
     * @param asOfDateTime Datetime used to retrieve the state of the configuration at that time. If null the current state will be retrieved.
```

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 7]
```diff
@@ -295,6 +279,9 @@
      * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.getSettingWithResponse#ConfigurationSetting}
      *
      * @param setting The setting to retrieve based on its key and optional label combination.
+     * @param asOfDateTime To access a past state of the configuration setting, or optionally, null if a setting with
+     * asOfDateTime is desired.
+     * @param ifChanged A boolean value indicates if using setting's ETag value to If-None-Match header.
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]:
```suggestion
     * @param ifChanged Flag indicating if the {@ code setting} {@link ConfigurationSetting#getETag ETag} is used as a If-None-Match header.
```

**Pull request merged**